lib/connection.js

```diff
@@ -3,10 +3,151 @@
 const common = require('metarhia-common');
 const EventEmitter = require('events');
 
+const parser = require('./parser.js');
+
+const BUFFER_BEGIN = 0;
+const INITIAL_BUFFER_SIZE = 16384;
+
+const encoding = {
```

Comment:
`ENCODING` it is collection of constants, so we will access it like `ENCODING.binary` or `ENCODING[encodingType]`